src/esp/io/JsonEspTypes.h

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#ifndef ESP_IO_JSONESPTYPES_H_
+#define ESP_IO_JSONESPTYPES_H_
+
+/** @file
+ * @brief See JsonAllTypes.h. Don't include this header directly in user code.
+ */
+
+#include "JsonBuiltinTypes.h"
+#include "JsonMagnumTypes.h"
+
+#include "esp/assets/RenderAssetInstanceCreationInfo.h"
+#include "esp/core/esp.h"
+#include "esp/gfx/replay/Keyframe.h"
+
+namespace esp {
+namespace io {
+
+inline JsonGenericValue toJsonValue(const esp::vec3f& vec,
+                                    JsonAllocator& allocator) {
+  JsonGenericValue floatsArray(rapidjson::kArrayType);
+  for (int i = 0; i < vec.size(); i++) {
+    floatsArray.PushBack(vec.data()[i], allocator);
+  }
+  return floatsArray;
+}
+
+inline bool fromJsonValue(const JsonGenericValue& floatsArray,
+                          esp::vec3f& vec) {
+  // TODO: helper for objects that are simply arrays, with error-handling
+  ASSERT(floatsArray.Size() == vec.size());
```

Comment:
Do we want to fail an assertion if array size is not correct? Or could we just return false with a message?  (That's what I would do before in the old jsonIntoVal functions.)

Reply:
I'll change to make this work similar to Magnum::Vector3.